var/spack/repos/builtin/packages/gmt/package.py

```diff
@@ -64,6 +64,12 @@ class Gmt(Package):
     patch('regexp.patch', when='@6.1.0')
     patch('type.patch', when='@4.5.9')
 
+    executables = ['^gmt-config$']
+
+    @classmethod
+    def determine_version(cls, exe):
+        return Executable(exe)('--version', output=str).rstrip()
```

Comment:
It's probably better to redirect also `error` either to string or to something else. The point is that if you pick a "wrong" executable to test (very unlikely in this recipe) stderr will be prompted verbatim to user, which looks ugly. For example, I got that for `openssl`:
```console
$ spack external find openssl 
Can't locate Debian/Debhelper/Dh_Lib.pm in @INC (you may need to install the Debian::Debhelper::Dh_Lib module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.26.1 /usr/local/share/perl/5.26.1 /usr/lib/x86_64-linux-gnu/perl5/5.26 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.26 /usr/share/perl/5.26 /usr/local/lib/site_perl /usr/lib/x86_64-linux-gnu/perl-base) at /usr/bin/dh_perl_openssl line 6.
BEGIN failed--compilation aborted at /usr/bin/dh_perl_openssl line 6.
```
because I was testing `dh_perl_openssl`.